components/fxa-client/src/push.rs

```diff
@@ -0,0 +1,139 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use crate::{error::*, AccountEvent, FirefoxAccount};
+use serde_derive::Deserialize;
+
+impl FirefoxAccount {
+    /// Handle any incoming push message payload coming from the Firefox Accounts
+    /// servers that has been decrypted and authenticated by the Push crate.
+    ///
+    /// Due to iOS platform restrictions, a push notification must always show UI,
+    /// and therefore we only retrieve 1 command per message.
```

Comment:
I know this comment already existed, but it took me a few seconds to remember why one fact lead to the other. It may be worth updating it to say "...a push notification must always show UI. Since FxA sends one push notification per commend received, we must only retrieve 1 command per push message, otherwise we risk receiving push messages for which the UI has already been shown" or similar.

I also wonder whether the comment would be better down in the code at the part where we `if cfg!(target_os = "ios")`, to make it more obvious that this restriction only applies on iOS. When I first re-read this comment, I thought it meant that *all* platforms had this restriction, which was forced on them by requirements from iOS.